Restructure DS website's information architecture

contolini opened the pull request:
This PR restructures the site's navigation and moves every page into its new category. There's no longer a nav in the header, it's all in the sidebar now. It's also editable via the CMS to let users place pages exactly where they want instead of everything getting listed alphabetically.

We used to have a netlify cms [collection](https://github.com/cfpb/design-system/blob/ac5ee5e838543eb38e33c9b71401c39e0e36c70f/docs/admin/config.yml#L23) for every category but IMHO that's silly so I deleted them all and created a new collection called `pages` that houses every page.

## Changes

- Removes main nav and moves all nav elements to left-hand sidebar.
- Adds a side nav [data file](https://jekyllrb.com/docs/datafiles/) that is editable via the CMS and generates the left-hand sidebar.
- Adds a netlify cms collection called "Navigation" for the aforementioned new side nav data file.
- Removes the four netlify cms collections that were identical and replaces them with a single one called "Pages". Accordingly, I moved all website pages into a folder called `pages/`.
- Moves special pages like the homepage and the help page to a new collection called "Special pages".
- Updated the "Updating this website" page with instructions on how to update stuff.
- Removed the "Add new page button" and replaced it with a link in the footer.
- Updates tests and Jekyll/netlify metadata files as appropriate

## Testing

1. `yarn test:browser` should pass
1. Click around the PR preview link below and see how you feel about the new all-inclusive side nav.
1. Go to the "Updating this website" page by clicking the link in the footer and peruse the instructions for creating new pages and editing the side nav. **Note:** If you try and edit the side nav in the CMS on the PR preview website, it will be blank in the CMS because the `side-navigation.yml` file doesn't exist in this repo's master branch. You can run the site locally to preview it.

## Screenshots

| new side nav | editing side nav in CMS | new footer links |
|--------------|-------------------------|------------------|
| <img alt="screenshot" src="https://user-images.githubusercontent.com/1060248/85316075-1dffa080-b48a-11ea-8780-7042c96149c8.png"> | <img alt="screenshot" src="https://user-images.githubusercontent.com/1060248/85316080-1fc96400-b48a-11ea-9a44-61850381e26a.png"> | <img alt="screenshot" src="https://user-images.githubusercontent.com/1060248/85316078-1f30cd80-b48a-11ea-9d0b-660b08187aaa.png"> |



## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

## Testing checklist

### Browsers

- [ ] Chrome on desktop
- [ ] Firefox
- [ ] Safari on macOS
- [ ] Edge
- [ ] Internet Explorer 9, 10, and 11
- [ ] Safari on iOS
- [ ] Chrome on Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

contolini commented:
I moved all the pages to a new folder so before merging this PR I'll manually remediate recent content conflicts.

anselmbradford commented on docs/_includes/sidebar.html:
```diff
@@ -1,46 +1,45 @@
+        </div>
+        {% if page.section == first_level_slug or page == blank %}
+        <div class="m-list_item__tertiary">
+            {% if first.nav-items != '' %}
```
I think this check needs to be more robust. I'm seeing this show up empty on the components page.

anselmbradford commented on docs/_includes/sidebar.html:
```diff
@@ -1,46 +1,45 @@
+            <a class="m-list_link" href="{{ first_level_slug | relative_url }}">{{ first.heading }}</a>
+        </div>
+        {% if page.section == first_level_slug or page == blank %}
+        <div class="m-list_item__tertiary">
```
Modifier should be used with a block or element to modify it. So this should be looking like 

```
m-list_item m-list_item__tertiary
```

…where `m-list_item` is the element that is adding some styles, and `m-list_item__tertiary` is the modifier changing some styles of that element.

This doesn't appear to me that you need to necessarily hook into the `m-list` architecture at all. If you did, this would be an element, like `m-list_item-subgroup` or similar, but if you didn't you could make this a utility class, `u-ml20` (for margin-left 20px, to fit with our existing utilities like this).

contolini commented:
>     * What if we put the sidenav on all pages, including the homepage? If not, why not?
>     * What if when you hovered over the sidenav it showed a little pencil icon like the edit this page buttton, and when you clicked that icon it brought you to the edit sidenav page?

🤯 🤯 🤯

contolini commented:
Thanks @anselmbradford. I fixed things and added the side nav to the homepage. Adding an edit button to the side nav is a great idea but beyond the scope of this PR.
